Jenkinsfile-asf

```
@@ -0,0 +1,77 @@
+#!groovy
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ * 
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+pipeline {
+    agent {
+        label 'cassandra-small'
+    }
+
+  triggers {
+    // schedules only run against release branches (i.e. 3.x, 4.x, 4.5.x, etc.)
+    cron(branchPatternCron().matcher(env.BRANCH_NAME).matches() ? "@weekly" : "")
+  }
+
+    stages {
+        stage('Matrix') {
+            matrix {
+                axes {
+                    axis {
+                        name 'TEST_JAVA_VERSION'
+                        values 'openjdk@1.8.0-292', 'openjdk@1.11.0-9', 'openjdk@17'
+                    }
+                    axis {
+                        name 'SERVER_VERSION'
+                        values '3.11.17',      
+				'4.0.13', 
+                                '5.0-beta1' 
```

```suggestion
                        values '3.11.17', '4.0.13', '5.0.0' 
```

otherwise, any way to make this dynamic ? 
e.g.
```
curl -s https://downloads.apache.org/cassandra/ | grep -oP '(?<=href=\")[0-9]+\.[0-9]+\.[0-9]+(?=)' | sort -rV | uniq -w 3
```

---

ccm does not work with `5.0.0` right now due to the memtable bug you reported
```
Caused by: org.apache.cassandra.exceptions.ConfigurationException: Configuration definition inherits unknown trie. A configuration can only extend one defined earlier or "default".
```
It only works with `5.0-beta1`. All the other drivers' CI also use `5.0-beta1` currently and have to change to `5.0.0` later when the above bug is fixed.

In terms of dynamic axis value, it is not allowed by design (see [this](https://stackoverflow.com/questions/59844846/declarative-pipeline-with-dynamic-matrix-axis-values)).

---

can you fix formatting at least please 🙏